### PR TITLE
CLI: Restore: SharePoint Pages

### LINKED ISSUE #2070: CLI:  SharePoint: --> Pages
-- Create CLI commands to be able to backup / restore `SharePoint.Pages`

### LINKED ISSUE #2168: CLI: SharePoint Selectors updates for restore
Finish SharePoint selectors for restore

### LINKED ISSUE #2218: CLI: Restore: SharePoint: Pages: Enable
Enable restore of `SharePoint.Pages` through the CLI.

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Changes connect SharePoint Pages to restore CLI. 

## Does this PR need a docs update or release note?
- [x] :no_entry: No Arguments are hidden

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* related to #2318 
* closes #2070 
* closes #2168<issue>
* closes #2218

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

**Comment by dadams39** [comment]
@ryanfkeepers , Verifying functionality

```bash
./corso restore sharepoint --backup <bupID> --page-item <pid>--log-level debug --hide-progress
Error: Failed to run SharePoint restore: 2 errors occurred:
	* errors prevented the operation from processing
	* restoring service data: restore of pages not supported
```
works as expected; however,
```bash
./corso restore sharepoint --backup <bupID> --page <pid>--log-level debug --hide-progress
Error: Failed to run SharePoint restore: 2 errors occurred:
	* errors prevented the operation from processing
	* retrieving service data: no restore path given
```

We should be able to restore all of the associated pages for the site within the collection with this command.

File: src/cli/restore/sharepoint.go [hunk 1]
```diff
@@ -76,6 +78,18 @@ func addSharePointCommands(cmd *cobra.Command) *cobra.Command {
...
 
+		fs.StringSliceVar(
+			&pageFolders,
+			utils.PageFN, nil,
+			"Restore Site pages by SharePoint Site ID",
+		)
+
+		fs.StringSliceVar(
+			&pages,
+			utils.PageItemFN, nil,
```
Comment (ryanfkeepers) [inline src/cli/restore/sharepoint.go:89]: The flag name should get updated to match the other formats.  Ie: the flag for page folders should be `page-folders`, and items should be `pages`.  Less confusion that way.

Comment (dadams39) [reply src/cli/restore/sharepoint.go]: Resolved with [5414c71](https://github.com/alcionai/corso/pull/2228/commits/5414c71581c3cd12499e4eb6cf934bbd78464a1f)

**Comment by ryanfkeepers** [comment]
@dadams39 This needs to be rebased before I'm able to review it.

File: src/internal/connector/graph/betasdk/beta_client.go [hunk 2]
```diff
@@ -15,6 +15,17 @@ import (
 // Details on how the Code was generated is present in `kioter-lock.json`.
 // NOTE: kiota gen file is altered to indicate what files are included in the created
 //
+<<<<<<< HEAD
```
Comment (ryanfkeepers) [inline src/internal/connector/graph/betasdk/beta_client.go:18]: leftover rebase conflict

File: src/cli/restore/sharepoint.go [hunk 3]
```diff
@@ -76,6 +78,18 @@ func addSharePointCommands(cmd *cobra.Command) *cobra.Command {
 			utils.ListItemFN, nil,
 			"Restore list items by ID")
 
+		fs.StringSliceVar(
+			&pageFolders,
+			utils.PageFolderFN, nil,
+			"Restore Site pages by SharePoint Site ID",
```
Comment (ryanfkeepers) [inline src/cli/restore/sharepoint.go:84]:
```suggestion
			"Restore Site pages by page folder name",
```

File: src/cli/backup/sharepoint.go [hunk 4]
```diff
@@ -140,12 +140,12 @@ func addSharePointCommands(cmd *cobra.Command) *cobra.Command {
 
 		fs.StringSliceVar(
 			&pageFolders,
-			utils.PageFN, nil,
+			utils.PageFolderFN, nil,
 			"Select backup data by site ID; accepts '"+utils.Wildcard+"' to select all sites.")
```
Comment (ryanfkeepers) [inline src/cli/backup/sharepoint.go:144]:
```suggestion
			"Select backup data by folder name; accepts '"+utils.Wildcard+"' to select all folders.")
```

**Review by ryanfkeepers — APPROVED** [review]
Some nits, and a leftover rebase conflict.  Approving to unblock.

File: src/internal/connector/graph/betasdk/beta_client.go (hunk 2 above)
Comment (dadams39) [reply src/internal/connector/graph/betasdk/beta_client.go]: Resolved with [4b001a2](https://github.com/alcionai/corso/pull/2228/commits/4b001a20a158b0685344be8305fd39e371d8650a)

File: src/cli/backup/sharepoint.go (hunk 4 above)
Comment (dadams39) [reply src/cli/backup/sharepoint.go]: Resolved with [72ff596](https://github.com/alcionai/corso/pull/2228/commits/72ff596d2aa6cef83d699431351433f58dd6f4f2)

File: src/cli/restore/sharepoint.go (hunk 3 above)
Comment (dadams39) [reply src/cli/restore/sharepoint.go]: Resolved within [3b2e1ec](https://github.com/alcionai/corso/pull/2228/commits/3b2e1ecdc20f30719b3f6fe44483a8c77f2df6e9)

File: src/internal/connector/graph/betasdk/beta_client.go (hunk 2 above)
Comment (dadams39) [reply src/internal/connector/graph/betasdk/beta_client.go]: Resolved within [4b001a2](https://github.com/alcionai/corso/pull/2228/commits/4b001a20a158b0685344be8305fd39e371d8650a)

**Pull request merged**